backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/database/service/AccessRecordDatabase.kt

```diff
@@ -0,0 +1,58 @@
+package com.tngtech.apicenter.backend.connector.database.service
+
+import com.tngtech.apicenter.backend.connector.database.entity.AccessRecordEntity
+import com.tngtech.apicenter.backend.connector.database.entity.AccessRecordId
+import com.tngtech.apicenter.backend.connector.database.repository.AccessRecordRepository
+import com.tngtech.apicenter.backend.domain.entity.PermissionType
+import com.tngtech.apicenter.backend.domain.entity.ServiceId
+import com.tngtech.apicenter.backend.domain.service.PermissionsManager
+
+@org.springframework.stereotype.Service
+class AccessRecordDatabase constructor(
+        private val accessRecordRepository: AccessRecordRepository
+): PermissionsManager {
+    override fun addPermission(userId: String, serviceId: ServiceId, permission: PermissionType) {
+        val key = AccessRecordId(serviceId.id, userId)
+        val record = accessRecordRepository.findById(key)
+
+        val view: Boolean = (permission == PermissionType.VIEW) ||
+            record.map { entity -> entity.view }.orElse(false)
+
+        val viewPrereleases: Boolean = (permission == PermissionType.VIEWPRERELEASE) ||
+            record.map { entity -> entity.viewPrereleases }.orElse(false)
+
+        val edit: Boolean = (permission == PermissionType.EDIT) ||
+            record.map { entity -> entity.edit }.orElse(false)
+
+        accessRecordRepository.save(AccessRecordEntity(key, view, viewPrereleases, edit))
+    }
+
+    override fun removePermission(userId: String, serviceId: ServiceId, permission: PermissionType) {
```

Comment:
I'm a bit confused. Do I specify which permissions to remove? Would a generic `setPermissions()` be better?